src/aca_main.cpp

```diff
@@ -261,15 +267,25 @@ int main(int argc, char *argv[])
   }
 
   g_grpc_server = new GoalStateProvisionerAsyncServer();
-  g_grpc_server_thread = new std::thread(std::bind(
-          &GoalStateProvisionerAsyncServer::RunServer, g_grpc_server, thread_pools_size));
-  g_grpc_server_thread->detach();
-
+  
   // Create a separate thread to run the grpc client.
   g_grpc_client = new GoalStateProvisionerClientImpl();
-  g_grpc_client_thread = new std::thread(
-          std::bind(&GoalStateProvisionerClientImpl::RunClient, g_grpc_client));
-  g_grpc_client_thread->detach();
+
+  // Create a marl scheduler using all the logical processors available to the process.
+  // Bind this scheduler to the main thread so we can call marl::schedule()
+  marl::Scheduler::Config cfg_bind_hw_cores;
+  cfg_bind_hw_cores.setWorkerThreadCount(thread_pools_size * 2);
```

Comment:
I made the worker thread count to be `thread_pool_size * 2`, instead of a fixed number.